packages/back-end/src/services/files.ts

```diff
@@ -65,6 +70,30 @@ export async function getFileUploadURL(ext: string, pathPrefix: string) {
   const filename = uniqid("img_");
   const filePath = `${pathPrefix}${filename}.${ext}`;
 
+  async function getSignedGoogleUrl() {
+    const projectId = GCS_PROJECT_ID;
+    const keyFilename = GCS_PRIVATE_KEY_FILEPATH;
+
+    const storage = new Storage({
+      projectId,
+      keyFilename,
+    });
```

Comment:
Google Cloud can be configured entirely by environment variables, so in code you can just do `new Storage()` - https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable

For testing, if you use the following environment variable names, they should get picked up automatically without you needing to pass them in the code.
- `GOOGLE_CLOUD_PROJECT`
- `GOOGLE_APPLICATION_CREDENTIALS` (path to the key file)

What's even cooler, is that if you deploy the GrowthBook container to Google Cloud, you don't even need environment variables.  It can do the authentication automatically for you.

Reply:
My mind is blown - that's way too cool.

In my testing, I found that I didn't even have to add a env variable for `GOOGLE_CLOUD_PROJECT` as that was in the key file. 

Is there a use case that I'm not aware of where we'd need to support the `GOOGLE_CLOUD_PROJECT` variable?

If not, then we just need to support `GCS_BUCKET_NAME` and `GOOGLE_APPLICATION_CREDENTIALS`.

I'll make a note for our next 1:1, but I'm curious how companies host the `GOOGLE_APPLICATION_CREDENTIALS` for a production site. Would they just host it in a private GCS bucket, and then provide the url when setting the env variable (if GrowthBook wasn't deployed to Google Cloud?

Reply:
I think you can make a key file that has permissions for multiple projects, so that env variable lets you customize it.  However, the people doing that will already know how to set the env variables, so we don't need documentation for it.  Will just confuse other people.